### PR TITLE
ALP-226 Generated visitor name

### PR DESCRIPTION
**Pull request by RamsesDijkstra**
Tested this with duplicate names, which will cause the generator to be re-called all seems good to go

### CODE REVIEW

File: database/users/db.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ const (
          						                	WHERE plugin_id = '86DC3AE7-9F3D-42CB-85A3-A71ABC3C3CB8'
          						                    AND attribute_name = 'wallet'
          						                    AND value->'wallet' ? $1);`
+	getUserByNameQuery          = `SELECT * FROM "user" WHERE profile->'name' ? $1;`
```
Comment (cnaize) [inline database/users/db.go]: maybe it's better to use `WHERE profile->>'name' = $1`?
I don't know the exact difference, but let's check? :)

Comment (cnaize) [reply database/users/db.go]: also I think it's better to write additional query only for existence testing, without getting the whole user data

Comment (cnaize) [reply database/users/db.go]: and the main question - what about decentralization?

File: universe/common/api/auth.go [hunk 2]
```diff
@@ -126,3 +127,22 @@ func ValidateJWTWithSecret(signedString string, secret []byte) (*jwt.Token, erro
 		return secret, nil
 	})
 }
+
+func GenerateVisitorName(c *gin.Context, db database.DB) (string, error) {
```
Comment (cnaize) [inline universe/common/api/auth.go]: maybe `GenerateGuestName()` is better?

File: database/users/db.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ const (
          						                	WHERE plugin_id = '86DC3AE7-9F3D-42CB-85A3-A71ABC3C3CB8'
          						                    AND attribute_name = 'wallet'
          						                    AND value->'wallet' ? $1);`
+	userExistsByNameQuery       = `SELECT EXISTS(SELECT * FROM "user" WHERE profile->>'name' = $1);`
```
Comment (cnaize) [inline database/users/db.go]: I'm not sure about Postgres optimizations, but maybe `... EXISTS(SELECT 1 ...` is better?

File: database/interfaces.go [hunk 4]
```diff
@@ -68,6 +68,8 @@ type UsersDB interface {
 	GetUserByWallet(ctx context.Context, wallet string) (*entry.User, error)
 	GetUserProfileByUserID(ctx context.Context, userID uuid.UUID) (*entry.UserProfile, error)
 
+	UserExistsByName(ctx context.Context, name string) (bool, error)
```
Comment (cnaize) [inline database/interfaces.go]: suggest to rename to `CheckIsUserExistsByName()`

**Review by terbooter — APPROVED** [review]

**Review by cnaize — APPROVED** [review]

**Pull request merged**